Added NS incompressible and Euler equation

chaous opened the pull request:
In some cases it is better to use incompressible NS and Euler equation uses way less memory

chaous commented:
I fixed tessellation, so now will work on windows via docker or pip. I used mesh_to_sdf instead of pysdf

michaltakac commented:
Looking forward for reviews, since our team cannot upgrade to modulus:23.05 due to missing pysdf library

cfd1 commented on README.md:
````diff
@@ -67,6 +67,22 @@ Alternatively, if you want to skip pysdf installation, you can run the following
+pip install trimesh
+pip install mesh_to_sdf
+```
+Tesselation will work because insted of using pysdf I used mesh_to_sdf (I have not yet implemented sdf derivatives yet, so some function will not work on windows (It's only a few functions for visualisation difference with real data (as far as I know)))
````
Minor typo, should be instead

cfd1 commented on modulus/sym/eq/pdes/navier_stokes.py:
```diff
@@ -339,6 +339,90 @@ def __init__(self, T, dim=3, time=True):
         )
 
+class NavierStokesIncompressible(PDE):
+    '''Cystom implementation of incompressible NS eqyastyans'''
```
Should be custom and equations

cfd1 commented on modulus/sym/eq/pdes/navier_stokes.py:
```diff
@@ -507,3 +591,84 @@ def __init__(self, T="T", D="D", rho=1, vec=["u", "v", "w"]):
+
+
+class Euler(PDE):
+    '''Cystom implementation of compressible Euler eqyastyans'''
```
Typo as above

cfd1 commented on modulus/sym/eq/pdes/navier_stokes.py:
```diff
@@ -339,6 +339,90 @@ def __init__(self, T, dim=3, time=True):
+
+        # density
+        if isinstance(rho, str):
+            rho = Function(rho)(*input_variables)
```
If this incompressible, then surely rho must be a constant? Therefore raising an exception would make more sense

chaous replied:
You probably correct, I will fix it

chaous commented:
Implemented all suggestions

cfd1 commented:
I need to check the code when I get a chance, but how is this different from calling the Navier Stokes with a constant rho?

chaous commented:
In classical CFD, for better results, incompressible form is used for low Mach numbers. I was able to achieve better results with an incompressible equation for PINNs than with a compressible equation but with constant rho

chaous commented on modulus/sym/eq/pdes/navier_stokes.py:
```diff
@@ -339,6 +339,86 @@ def __init__(self, T, dim=3, time=True):
+
+        # density
+        if isinstance(rho, str):
+            raise Exception("rho myst be number")
```
fixed

cfd1 commented:
Just for the record I can’t review or merge your work.

cfd1 commented:
If you can it would be good to have an example of your changes working. You could use the lid driven cavity for the incompressible formulation. I’m not sure if there is a compressible example with shockwaves to show off the Euler equations.

YouqiongLiu commented:
> If you can it would be good to have an example of your changes working. You could use the lid driven cavity for the incompressible formulation. I’m not sure if there is a compressible example with shockwaves to show off the Euler equations.

Can you work via pip?

YouqiongLiu commented:
> I fixed tessellation, so now will work on windows via docker or pip. I used mesh_to_sdf instead of pysdf

Added  incompressible NS and fixed tessellation，Can the aneurysm.py file run？

YouqiongLiu commented:
> I fixed tessellation, so now will work on windows via docker or pip. I used mesh_to_sdf instead of pysdf

e.g  modulus-sym/examples/geometry/tessellated_example.py

YouqiongLiu commented:
> I fixed tessellation, so now will work on windows via docker or pip. I used mesh_to_sdf instead of pysdf

e.g modulus-sym\examples\geometry\tessellated_example.py

chaous commented:
> > I fixed tessellation, so now will work on windows via docker or pip. I used mesh_to_sdf instead of pysdf
> 
> e.g modulus-sym\examples\geometry\tessellated_example.py your fixed tessellation can work, but the following warning appears. /modulus/sym/geometry/tessellation.py:179: RuntimeWarning: divide by zero encountered in divide np.full(x.shape, triangle_areas[index] / x.shape[0])

It appears on on regular tessellation too. This warning can be ignored

ktangsali commented:
Hi @chaous , sorry for the delay in getting attention to this PR. Thanks for the contribution! Can we split this PR into two? One that adds equations and the other that adds changes to tessellation? Reviewing the tessellation might take some time, but we can get the equations added much sooner. Also, if you can add an example to go along with your equation additions that can demonstrate potential speed-up/improvement in accuracy, that would be helpful.

chaous commented:
> Hi @chaous , sorry for the delay in getting attention to this PR. Thanks for the contribution! Can we split this PR into two? One that adds equations and the other that adds changes to tessellation? Reviewing the tessellation might take some time, but we can get the equations added much sooner. Also, if you can add an example to go along with your equation additions that can demonstrate potential speed-up/improvement in accuracy, that would be helpful.

I split my pull request into two and added examples 
Equations https://github.com/NVIDIA/modulus-sym/pull/142
tessellation https://github.com/NVIDIA/modulus-sym/pull/143
